.travis.yml

```diff
@@ -13,20 +13,8 @@ before_install:
 install: true # skips travis' default installation step which executes gradle assemble.
 jobs:
   include:
-    - stage: benchmark#1
-      if: tag IS present
-      script: ./gradlew clean setLibraryVersion benchmark benchmarkCommit
-    - stage: benchmark#2
-      if: tag IS present
-      script: ./gradlew clean setLibraryVersion benchmark benchmarkCommit
-    - stage: benchmark#3
-      if: tag IS present
-      script: ./gradlew clean setLibraryVersion benchmark benchmarkCommit
-    - stage: benchmark#4
-      if: tag IS present
-      script: ./gradlew clean setLibraryVersion benchmark benchmarkCommit
-    - stage: benchmark#5
-      if: tag IS present
+    - stage: benchmark
```

Comment:
This line is basically setting a name for the stage -> `benchmark`.

`if: branch = master AND commit_message =~ /^(Merge pull request .*)/` : this condition means only run this stage when the branch is master on the commit message contains this regular expression.

Reply:
Why do you check additionally for commit_message?

Reply:
I wanted to make sure that the commit is from a merge of a PR to master.

Reply:
Why not? How often do we need to commit to master directly?

Reply:
Rarely, almost never. But when it happens it's usually a (quick typo fix, etcc) (basically sth that doesn't need a review). Otherwise, a PR has to be opened.